frontend/packages/topology/src/utils/useHover.ts

```diff
@@ -25,21 +26,27 @@ const useHover = <T extends Element>(
           // store locally instead of a ref because it only needs to be referred by inner closures
           let delayHandle: any;
 
-          const onMouseEnter = () => {
-            if (delay != null) {
+          const delayedStateChange = (newState: boolean) => {
+            clearTimeout(unsetHandle.current);
+            clearTimeout(delayHandle);
+
+            if (timeoutDelay != null) {
               delayHandle = window.setTimeout(() => {
-                delayHandle = clearTimeout(unsetHandle.current);
-                setHover(true);
-              }, delay);
+                clearTimeout(unsetHandle.current);
```

Comment:
Trying to recall why this is inside the timeout and not outside. However by you clearing the timeout above, we don't need this here.

Reply:
it is needed if the timer is created during the hover delay, not needed below when there is no delay.